ruby/lib/ci/queue/redis/base.rb

```diff
@@ -8,6 +8,7 @@ class Base
         TEN_MINUTES = 60 * 10
         CONNECTION_ERRORS = [
           ::Redis::BaseConnectionError,
+          RedisClient::ConnectionError,
```

Comment:
@casperisfine because we're using `without_reconnect` it calls `ensure_connected` on `RedisClient` and hence it throws a `RedisClient::ConnectionError` and not a `::Redis::BaseConnectionError`.

Are we fine including that one here?

Reply:
That means requiring `redis-rb` 5.x. But `redis-rb` shouldn't raise `RedisClient` errors, this seems like a bug to me.

Reply:
I suppose you can do:

```ruby
CONNECTION_ERRORS  << RedisClient::ConnectionError if defined?(RedisClient::ConnectionError)
```

Reply:
https://github.com/redis/redis-rb/pull/1312